app/src/main/java/com/example/dogbook/main/models/Post.java

```diff
@@ -145,4 +169,73 @@ public String getTitle() {
     public String getSnippet() {
         return this.getDescription();
     }
+
+    public void like(Context context, PostReactionCallback postReactionCallback) {
+        this.likedLocally();
+        postReactionCallback.onOptimisticUpdate(this);
+
+        //Apply changes on database
+        Like like = new Like();
+        like.setAuthor(ParseUser.getCurrentUser());
+        like.setPost(this);
+        like.saveInBackground(new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                if (e == null) {
+                    Log.i(TAG, "Like saved successfully");
+                    postReactionCallback.onSuccess(Post.this);
+                    return;
+                }
+                Log.e(TAG, "Issue saving Like", e);
+                Toast.makeText(context, "Liking was not possible", Toast.LENGTH_SHORT).show();
+                Post.this.dislikedLocally();
+                postReactionCallback.onFailure(Post.this);
+            }
+        });
+
+    }
```

Comment:
Hm, I think this and dislike would be a bit better if they lived in a data specific class instead of the model class.  For the model class, we should try to keep it smaller and single purposed -- kind of how you handled it for Twitter.  It seems like you're doing it here in order to keep the like/dislike locally private, but I think it would be a bit cleaner if some class/method was passed a post object, context, + method and performed the function calls instead of doing it in the model class